luigi/contrib/s3.py

```diff
@@ -163,350 +161,217 @@ def remove(self, path, recursive=True):
-            results = []
-
-            # Calculate the number of chunks the file will be
-            num_parts = (key_size + part_size - 1) // part_size
-
-            for i in range(num_parts):
-                # Issue an S3 copy request, one part at a time, from one S3 object to another
-                part_start = cur_pos
-                cur_pos += part_size
-                part_end = min(cur_pos - 1, key_size - 1)
-                part_num = i + 1
-                results.append(pool.apply_async(mp.copy_part_from_key, args=(src_bucket, src_key, part_num, part_start, part_end)))
-                logger.info('Requesting copy of %s/%s to %s/%s', part_num, num_parts, dst_bucket, dst_key)
-
-            logger.info('Waiting for multipart copy of %s/%s to finish', dst_bucket, dst_key)
-
-            # This will raise any exceptions in any of the copy threads
-            for result in results:
-                result.get()
-
-            # finish the copy, making the file available in S3
-            mp.complete_upload()
-            return mp.key_name
-
-        except:
-            logger.info('Error during multipart s3 copy for %s/%s to %s/%s...', src_bucket, src_key, dst_bucket, dst_key)
-            # cancel the copy so we don't get charged for storage consumed by copied parts
-            if mp:
-                mp.cancel_upload()
-            raise
-
-    def move(self, source_path, destination_path, **kwargs):
-        """
-        Rename/move an object from one S3 location to another.
+        if not parents and not self.isdir(os.path.dirname(path)):
+            raise MissingParentDirectory()
 
-        :param kwargs: Keyword arguments are passed to the boto function `copy_key`
-        """
-        self.copy(source_path, destination_path, **kwargs)
-        self.remove(source_path)
+        return self.put_string("", path)
 
     def listdir(self, path, start_time=None, end_time=None, return_key=False):
         """
         Get an iterable with S3 folder contents.
         Iterable contains paths relative to queried path.
-
         :param start_time: Optional argument to list files with modified dates after start_time
         :param end_time: Optional argument to list files with modified dates before end_time
         :param return_key: Optional argument, when set to True will return a boto.s3.key.Key (instead of the filename)
         """
         (bucket, key) = self._path_to_bucket_and_key(path)
 
         # grab and validate the bucket
-        s3_bucket = self.s3.get_bucket(bucket, validate=True)
+        s3_bucket = self.s3.Bucket(bucket)
 
         key_path = self._add_path_delimiter(key)
         key_path_len = len(key_path)
-        for item in s3_bucket.list(prefix=key_path):
-            last_modified_date = time.strptime(item.last_modified, "%Y-%m-%dT%H:%M:%S.%fZ")
+        for item in s3_bucket.objects.filter(Prefix=key_path):
+            last_modified_date = item.last_modified.date()
```

Comment:
This removes the ability to limit dir contents by datetime.

When I added this functionality, i only needed it for dates. But perhaps someone may need it for more granular times.